### ISSUE TITLE
Allow Style/AlignHash with key alignment to remove extra spacing before values

### ISSUE THREAD

**Issue by jzinn** [rubocop-hq/rubocop#3727]
When `Style/AlignHash` is configured for key alignment, the space between keys and values is ignored.  Please add the capability to remove extra spacing between keys and values.

## Expected behavior

With this configuration file:
```yaml
Style/AlignHash:
  EnforcedHashRocketStyle: key
  EnforcedColonStyle: key

Style/ExtraSpacing:
  AllowForAlignment: false
```

And this test ruby file:
```ruby
{
  a:  :b,
  c: :d
}
```

I would expect the file to be autocorrected to this:
```ruby
{
  a: :b,
  c: :d
}
```

## Actual behavior

The file is unchanged.

## RuboCop version

```
$ rubocop -V
0.45.0 (using Parser 2.3.1.4, running on ruby 2.3.2 x86_64-darwin15)
```

**Comment by Drenmi** [comment]
`Style/ExtraSpacing` intentionally ignores hashes and defers their inspection to `Style/AlignHash`, but this is still a reasonable feature request to `Style/AlignHash`. 😀

**Comment by dbackeus** [comment]
+1 for fixing `Style/AlignHash` with `key`.

It currently accepts `table` formatting even though the docs makes it look like it should enforce keeping everything left-aligned: https://github.com/bbatsov/rubocop/blob/master/config/default.yml#L155
